### ISSUE TITLE
Tooltips in toolbar not helpful

### ISSUE THREAD

**Issue by buhtz** [bit-team/backintime#1605]
For beginner contributors looking for a _Good First Issue_: See [comment in this discussion](https://github.com/bit-team/backintime/issues/1605#issuecomment-1893537459) for instructions how to solve the issue.

The tooltips in our mainwindow are connected to `QActions` instead of specific buttons and menu entries. These actions are then connected to buttons and menu entries. The consequence is that menu entries and buttons in the toolbar do share the same tooltip.

![Peek 2024-01-08 15-05](https://github.com/bit-team/backintime/assets/11861496/e33a8456-4c9f-4922-8187-6cbcad9e42d0)

The menu entry "Take a snapshot" describe itself by its label. The tooltip then is used to give background or more detailed information.
The toolbar button (usually) do not have a label string. Because of that most (?) users do expect that the tooltip of such a button do describe its function in a short way. The same way a menu entry label would do.

In short: The tooltip of the button should be "Take a snapshot" or something else.

There are some solutions we could think about. It is not urgent. I just wanted to document it.

**Comment by aryoda** [comment]
> In short: The tooltip of the button should be "Take a snapshot" or something else.

Perhaps our general pattern for tooltips should be:

    _<action name>: <details>_

In above case:

Instead of 

     `Use modification time & size for file change detection.`

better

    `Take a snapshot: Use modification time & size for file change detection.`

Then it does always work for GUI widgets that do **not** show a label but only an icon...

**Comment by buhtz** [comment]
OK, I support the approach of aryoda to modify the tooltip using the template `_<action name>: <details>_`.

The relevant tooltips can be found in [`app.py::MainWindow._create_actions()` method](https://github.com/bit-team/backintime/blob/136f466a88606dfb6eaa112f29a8531511531c68/qt/app.py#L427). To know which of these "actions" are relevant for the main windows toolbar have a look into [`app.py::MainWindow._create_main_toolbar()`](https://github.com/bit-team/backintime/blob/136f466a88606dfb6eaa112f29a8531511531c68/qt/app.py#L681).

**Comment by buhtz** [comment]
I might have an alternative approach that needs further investigation.
Here is some pseudo code to give an idea.

```
# The action object do has a text (label) and a tooltip
txt = act_take_snapshot.text()  # "Take snapshot"
tip = act_take_snapshot.toolTip()  # "Use modification time & size for file change detection."

# ...

# Use them to override the tooltip when creating the toolbar button
for act in actions_for_toolbar:
    toolbar.addAction(act)

    # if act has text & tooltip ...
    button_tip = f'{act.text()}: {act.toolTip()}'

    # Get the button widget
    button = toolbar.widgetForAction(act)
    button.setToolTip(button_tip)
```

Advantages are 
 - The source strings don't need to be changed.
 - The tooltip of the menu entry don't contain a redundant string (.e.g. "Take snapshot") that still exists in the label of the menu entry.

Possible disadvantage is that languages with reading directions "right to left" having the button tooltip in the wrong order. It should be "tooltip :label". But users with such languages are suited to software with such errors and we could check for languages like that. I assume that Qt itself knows its current reading direction. Otherwise we could check the current local/language settings explicit.

**EDIT**:
[`QWidget::layoutDirection()`](https://www.riverbankcomputing.com/static/Docs/PyQt6/api/qtwidgets/qwidget.html#layoutDirection)
